spec/models/endpoint_spec.rb

```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+
+RSpec.describe Endpoint, type: :model do
+
+  let!(:endpoint) { Endpoint.create(endpoint_name: 'aws', endpoint_type: 'cloud') }
+
+  it 'is not valid without valid attributes' do
+    expect(Endpoint.new).not_to be_valid
+  end
+
+  it 'is valid with valid attributes' do
+    expect(endpoint).to be_valid
+  end
+
+  it 'enforces unique constraint on endpoint_name' do
+    expect do
+      Endpoint.create!(endpoint_name: 'aws', endpoint_type: 'cloud')
```

Comment:
How about changing the endpoint_type to be something other than 'cloud' just to emphasize the 'endpoint_name' requirement?

Reply:
i think we're already going to revisit this (per #46) and turn `endpoint_type` into an FK to a table that has richer endpoint info.  so i'm inclined to leave this as-is, just to not tweak something that'll be changing in a different way anyway.

Reply:
👍